module/apmgopg/hook_test.go

```diff
@@ -0,0 +1,93 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build go1.9
+
+package apmgopg_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/go-pg/pg"
+	"github.com/go-pg/pg/orm"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.elastic.co/apm/apmtest"
+)
+
+type User struct {
+	TableName struct{} `sql:"users"`
+
+	Id   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+func TestWithContext(t *testing.T) {
+	if os.Getenv("PGHOST") == "" {
+		t.Skip("PGHOST env not defined, skipping")
+		return
+	}
+
+	_, spans, errors := apmtest.WithTransaction(func(ctx context.Context) {
+		db := pg.Connect(&pg.Options{
+			Addr:     fmt.Sprintf("%s:5432", os.Getenv("PGHOST")),
+			User:     "postgres",
+			Password: "hunter2",
+			Database: "test_db",
+		})
+
+		_, err := db.Exec("SELECT 1")
+		require.NoError(t, err)
+
+		db.CreateTable(&User{}, &orm.CreateTableOptions{})
+
+		defer db.Close()
+		db = db.WithContext(ctx)
+
+		_, err = db.Model(&User{Id: 1337, Name: "Antoine Hedgecock"}).Insert()
+		assert.NoError(t, err)
+
+		assert.NoError(t, db.Model(&User{}).Where("id = ?", 1337).Select())
+
+		_, err = db.Model(&User{Id: 1337, Name: "new name"}).WherePK().Update("Name")
+		assert.NoError(t, err)
```

Comment:
This fails with the error `pg: Model(non-pointer string)`

I think maybe this should be

```go
_, err = db.Model(&User{Id: 1337, Name: "new name"}).Column("name").WherePK().Update()
```